app/org/sagebionetworks/bridge/services/UploadService.java

```diff
@@ -344,8 +345,13 @@ public void uploadComplete(StudyIdentifier studyId, UploadCompletionClient compl
         ObjectMetadata obj;
         try {
             obj = s3Client.getObjectMetadata(uploadBucket, objectId);
-        } catch (AmazonClientException ex) {
-            throw new NotFoundException(ex);
+        } catch (AmazonS3Exception ex) {
+            if (ex.getStatusCode() == 404) {
+                throw new NotFoundException(ex);
+            } else {
+                // Only S3 404s are mapped to 404s. Everything else is an internal server error.
+                throw new BridgeServiceException(ex);
+            }
```

Comment:
Approved. I wonder if we could use a generic mapping class to convert all AWS constraint violation exceptions, all AWS not found exceptions, etc. in one place, so if fixed, it would be fixed everywhere in our codebase where it could occur.